nidaqmx/_grpc_stub_interpreter.py

```diff
@@ -0,0 +1,394 @@
+import threading
+import warnings
+
+import grpc
+import nidaqmx.nidaqmx_pb2 as grpc_types
+import nidaqmx.nidaqmx_pb2_grpc as nidaqmx_grpc
+import nidaqmx.session_pb2 as session_grpc_types
+
+class GrpcStubInterpreter(object):
+    '''Interpreter for interacting with a gRPC Stub class'''
+
+    def __init__(self, grpc_options):
+        self._grpc_options = grpc_options
+        self._lock = threading.RLock()
+        self._client = nidaqmx_grpc.NiDAQmxStub(grpc_options.grpc_channel)
+        self.session_initialization_behavior = grpc_options.initialization_behavior
+        self.set_task_handle()
+        self.set_system_handle()
+        self.set_scale_handle()
+
+    def set_task_handle(self, value=session_grpc_types.Session()):
+        self._task_handle = value
+
+    def get_task_handle(self):
+        return self._task_handle
+    
+    def set_system_handle(self, value=session_grpc_types.Session()):
+        self._system_handle = value
+
+    def get_system_handle(self):
+        return self._system_handle
```

Comment:
System properties & functions do not use handles. They use device/physchan/terminal names or no names at all. I think the Device and PhysicalChannel classes (not the interpreter class) should store the device/physchan names.

```
/******************************************************/
/***                Signal Routing                  ***/
/******************************************************/


int32 __CFUNC     DAQmxConnectTerms              (const char sourceTerminal[], const char destinationTerminal[], int32 signalModifiers);
int32 __CFUNC     DAQmxDisconnectTerms           (const char sourceTerminal[], const char destinationTerminal[]);
int32 __CFUNC     DAQmxTristateOutputTerm        (const char outputTerminal[]);


/******************************************************/
/***                Device Control                  ***/
/******************************************************/


int32 __CFUNC     DAQmxResetDevice               (const char deviceName[]);

int32 __CFUNC     DAQmxSelfTestDevice            (const char deviceName[]);

int32 __CFUNC_C   DAQmxGetDeviceAttribute        (const char deviceName[], int32 attribute, void *value, ...);
```
 
```
message ConnectTermsRequest {
  string source_terminal = 1;
  string destination_terminal = 2;
  oneof signal_modifiers_enum {
    InvertPolarity signal_modifiers = 3;
    int32 signal_modifiers_raw = 4;
  }
}

message ConnectTermsResponse {
  int32 status = 1;
}
```

```
message GetDeviceAttributeBoolRequest {
  string device_name = 1;
  oneof attribute_enum {
    DeviceBoolAttribute attribute = 2;
    int32 attribute_raw = 3;
  }
}

message GetDeviceAttributeBoolResponse {
  int32 status = 1;
  bool value = 2;
}
```

Reply:
That makes sense, i will remove this system handle local variable from the interpreters.